src/parser/mod.rs

```diff
@@ -2780,6 +2780,27 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// If the current token is the `expected` keyword followed by
+    /// specified tokens, consume them and returns true.
+    /// Otherwise, no tokens are consumed and returns false.
+    pub fn parse_keyword_with_tokens(&mut self, expected: Keyword, tokens: &[Token]) -> bool {
+        match self.peek_token().token {
+            Token::Word(w) if expected == w.keyword => {
+                for (idx, token) in tokens.iter().enumerate() {
+                    if self.peek_nth_token(idx + 1).token != *token {
```

Comment:
This is probably a non-issue since `tokens` should always be small, but it seems worth noting that this is unnecessarily quadratic over the length of `tokens` due to the loop in `peek_nth_token`.

Reply:
Thanks. Added a note for that.